### PR TITLE
Use `std::numeric_limits` instead of macros

### PR DESCRIPTION
**Pull request by hyperupcall**
This improves consistency with the rest of the codebase. I think this also improves compatability, since it's in #4092 as well

### CODE REVIEW

**Review by rolandlo — CHANGES_REQUESTED** [review]
Can you please apply "Include what you use" for the `<liimts>` header in the sense of https://include-what-you-use.org/ ? For instance it is currently neither included in `EditSelectionContents.cpp` nor `EditSelectionContents.h`.

Otherwise LGTM.

**Comment by hyperupcall** [comment]
Sorry about that - fixed now

File: src/core/plugin/Plugin.cpp [hunk 1]
```diff
@@ -2,6 +2,7 @@
 
 #include <algorithm>  // for max
 #include <array>      // for array
+#include <limits>     // for numeric_limits
```
Comment (rolandlo) [inline src/core/plugin/Plugin.cpp:5]: Already included in `Plugin.h`
```suggestion
```

**Review by rolandlo — APPROVED** [review]
I have one minor nitpick. Could you please squash the commits then into one and rebase on current master afterwards?

**Comment by hyperupcall** [comment]
:+1:

**Review by rolandlo — APPROVED** [review]
LGTM. 
Squashing and merging in 24 hours if no objections are raised.

**Comment by rolandlo** [comment]
Thanks for this PR as well! Always good to improve consistency.

**Pull request merged**